runners/katacoda/index.ts

```diff
@@ -186,17 +186,35 @@ export class Katacoda extends Runner {
         let fileDir = path.join(workspaceDir, runCommand.command.parameters[0]).replace(/\\/g, "/");
         let placeholder = runCommand.command.parameters[1].placeholder ? runCommand.command.parameters[1].placeholder : "";
         let dataTarget = runCommand.command.parameters[1].placeholder ? "insert" : "replace";
-
         let content = "";
+
+        if(runCommand.command.parameters[1].lineNumber)
+        {
+            let BGscript = "#!/bin/sh \n  \n sed 'lineNumber!#PLACEHOLDER#!' textdatei -i";
+            let number = parseInt(runCommand.command.parameters[1].lineNumber, 10);
+            let numberStr = (number == 1) ? (number.toString()+"i") : ((number-1).toString()+"a");
+            BGscript =BGscript.replace("lineNumber", numberStr);
+            BGscript =BGscript.replace("textdatei",fileDir);
+            let FGscript = fs.readFileSync(path.join(this.getRunnerDirectory(), "templates", "scripts", "insert_foreground.sh"), {encoding: "utf-8"});
+            FGscript = FGscript.replace("filename", runCommand.command.parameters[0]);
+            fs.writeFileSync(path.join(this.getRunnerDirectory(),"templates","scripts", "insert_background.sh"),BGscript, {encoding : "utf-8"});
+            fs.writeFileSync(path.join(this.getRunnerDirectory(),"templates","scripts", "insert_foreground.sh"),FGscript, {encoding : "utf-8"});
+
+            this.renderTemplate(path.join("scripts", "insert_background.sh"), path.join(this.outputPathTutorial, "insert_background.sh"), { });
+            this.renderTemplate(path.join("scripts", "insert_foreground.sh"), path.join(this.outputPathTutorial, "insert_foreground.sh"), { });
```

Comment:
If you have multiple changeFile steps, then each step would create a file with the same name. This wouldn't work. You have to differentiate between the file (e.g. add the step number)